# Assignment 1 : Python Data Analysis using salaries dataset

# Data Exploration:  This should include summary statistics, means, medians, quartiles, or any other relevant information about the data set. 

In [60]:
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [61]:
df = pd.read_csv('https://raw.githubusercontent.com/CunyLaguardiaDataAnalytics/datasets/master/Salaries.csv')

In [62]:
type(df)

pandas.core.frame.DataFrame

In [63]:
# 397 rows = 397 employees
df.shape

(397, 7)

In [64]:
df.head()

Unnamed: 0      rank discipline  yrs.since.phd  yrs.service   sex  salary
0           1      Prof          B             19           18  Male  139750
1           2      Prof          B             20           16  Male  173200
2           3  AsstProf          B              4            3  Male   79750
3           4      Prof          B             45           39  Male  115000
4           5      Prof          B             40           41  Male  141500

In [65]:
df.tail()

Unnamed: 0      rank discipline  yrs.since.phd  yrs.service   sex  salary
392         393      Prof          A             33           30  Male  103106
393         394      Prof          A             31           19  Male  150564
394         395      Prof          A             42           25  Male  101738
395         396      Prof          A             25           15  Male   95329
396         397  AsstProf          A              8            4  Male   81035

In [66]:
df.describe()

Unnamed: 0  yrs.since.phd  yrs.service         salary
count  397.000000     397.000000   397.000000     397.000000
mean   199.000000      22.314861    17.614610  113706.458438
std    114.748275      12.887003    13.006024   30289.038695
min      1.000000       1.000000     0.000000   57800.000000
25%    100.000000      12.000000     7.000000   91000.000000
50%    199.000000      21.000000    16.000000  107300.000000
75%    298.000000      32.000000    27.000000  134185.000000
max    397.000000      56.000000    60.000000  231545.000000

# Mean/median of all salaries

In [67]:
# Mean of all salaries
df["salary"].mean()

113706.45843828715

In [68]:
# Median of all salaries
df["salary"].median()

107300.0

# Median salary between gender and/or rank(title) Etc...


Making a new Data Frame that only has the columns 'sex' and 'salary'

In [69]:
sex_salary = df[["sex","salary"]]
sex_salary

sex  salary
0    Male  139750
1    Male  173200
2    Male   79750
3    Male  115000
4    Male  141500
..    ...     ...
392  Male  103106
393  Male  150564
394  Male  101738
395  Male   95329
396  Male   81035

[397 rows x 2 columns]

Finding the Median salary for all males

In [70]:
male = sex_salary[sex_salary['sex']=='Male']
male_median = male['salary'].median()
print(male_median)



108043.0


Finding the Median salary for all females

In [71]:
female = sex_salary[sex_salary['sex']=='Female']
female_median = female['salary'].median()
print(female_median)

103750.0


Comparing the Mean salary of Males and Females

In [72]:
mean_male = male['salary'].mean()
mean_male

115090.41899441341

In [73]:
mean_female = female['salary'].mean()
mean_female

101002.41025641025

In [74]:
# The difference between mean salaries of male and female
mean_male - mean_female

14088.008738003162

# Data Wrangling – perform some basic transformations on the data

People who have 20 years of experience (Total: 157), (Mean Salary: 122,520.66)

In [75]:
exp_20 = df[df['yrs.service'] >= 20]
print(exp_20.count())
mean_exp_20 = exp_20['salary'].mean()
print(mean_exp_20)

Unnamed: 0       157
rank             157
discipline       157
yrs.since.phd    157
yrs.service      157
sex              157
salary           157
dtype: int64
122520.65605095541


Comparing the salaries of people who have 20+ years of experience vs less than 20 years of experience

In [76]:
exp_less_20 = df[df['yrs.service'] < 20]
print(exp_less_20.count())
mean_exp_less_20 = exp_less_20['salary'].mean()
print(mean_exp_less_20)

Unnamed: 0       240
rank             240
discipline       240
yrs.since.phd    240
yrs.service      240
sex              240
salary           240
dtype: int64
107940.50416666667


In [77]:
mean_exp_20 - mean_exp_less_20

14580.151884288745

People who have salary of greater than or equal to 200,000

In [78]:
salary_200000 = df[df['salary'] >= 200000]
salary_200000.count()


Unnamed: 0       3
rank             3
discipline       3
yrs.since.phd    3
yrs.service      3
sex              3
salary           3
dtype: int64

Your Findings here...
There are only 3 people with a salary higher than 200000.

# Rename values, subset columns, etc

# Create some simple visualizations showing some simple statistics

In [79]:
print(df['salary'].max())
print(df['salary'].min())
print(df['salary'].mean())

231545
57800
113706.45843828715


Your observations from the visuals: The highest salary is 231545, the lowest salary is 57800, and the average salary is 113706.458.

# Draw some conclusions based on the data.  What did you discover?

Based on the analysis performed above, the conclusions are as follows:

Your Findings here...
There are 397 rows and 7 columns in the dataset. 


I also discovered that... The average salary of male workers is higher than the average salary of female workers.

Your Findings here... 
The average salary of people who work more than 20 years is higher than the average salary of people who work less than 20 years.



